spark/src/test/scala/org/apache/spark/sql/delta/clustering/ClusteredTableClusteringSuite.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.clustering
+
+import org.apache.spark.sql.delta.skipping.ClusteredTableTestUtils
+import org.apache.spark.sql.delta.skipping.clustering.ClusteredTableUtils
+import org.apache.spark.sql.delta.DeltaLog
+import org.apache.spark.sql.delta.actions.AddFile
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+
+class ClusteredTableClusteringSuite extends SparkFunSuite
+  with SharedSparkSession
+  with ClusteredTableTestUtils
+  with DeltaSQLCommandTest {
+  import testImplicits._
+
+  private val table: String = "test_table"
+
+  // Ingest data to create numFiles files with one row in each file.
+  private def addFiles(table: String, numFiles: Int): Unit = {
+    val df = (1 to numFiles).map(i => (i, i)).toDF("col1", "col2")
+    withSQLConf(SQLConf.MAX_RECORDS_PER_FILE.key -> "1") {
+      df.write.format("delta").mode("append").saveAsTable(table)
+    }
+  }
+
+  private def getFiles(table: String): Set[AddFile] = {
+    val deltaLog = DeltaLog.forTable(spark, TableIdentifier(table))
+    deltaLog.update().allFiles.collect().toSet
+  }
+
+  private def assertClustered(files: Set[AddFile]): Unit = {
+    assert(files.forall(_.clusteringProvider.contains(ClusteredTableUtils.clusteringProvider)))
+  }
+
+  test("optimize clustered table") {
+    withSQLConf(SQLConf.MAX_RECORDS_PER_FILE.key -> "2") {
+      withClusteredTable(
+        table = table,
+        schema = "col1 int, col2 int",
+        clusterBy = "col1, col2") {
+        addFiles(table, numFiles = 4)
+        val files0 = getFiles(table)
+        assert(files0.size === 4)
```

Comment:
can you assert that none of these files are clustered? (no clustering provider tag set)

Reply:
Done